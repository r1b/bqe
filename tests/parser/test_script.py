import pytest


@pytest.mark.parametrize(
    "sql",
    (
        pytest.param("SELECT col1 FROM table1;", id="script_single"),
        pytest.param("SELECT col1 FROM table1; SELECT col2 FROM table2", id="script_multi"),
        pytest.param(
            "SELECT col1 FROM table1; SELECT col2 FROM table2;",
            id="script_multi_with_trailing_semi",
        ),
        pytest.param("FROM table1;", id="script_top_level_from"),
    ),
)
def test_script_ok(sql, assert_parse_tree):
    assert_parse_tree(sql)
