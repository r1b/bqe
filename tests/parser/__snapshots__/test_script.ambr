# serializer version: 1
# name: test_script_ok[script_multi]
  '''
  query
    select
      select_list
        select_expr
          ident	col1
      from_clause
        from_item_table
          path_expression
            ident	table1
    select
      select_list
        select_expr
          ident	col2
      from_clause
        from_item_table
          path_expression
            ident	table2
  
  '''
# ---
# name: test_script_ok[script_multi_with_trailing_semi]
  '''
  query
    select
      select_list
        select_expr
          ident	col1
      from_clause
        from_item_table
          path_expression
            ident	table1
    select
      select_list
        select_expr
          ident	col2
      from_clause
        from_item_table
          path_expression
            ident	table2
  
  '''
# ---
# name: test_script_ok[script_single]
  '''
  select
    select_list
      select_expr
        ident	col1
    from_clause
      from_item_table
        path_expression
          ident	table1
  
  '''
# ---
# name: test_script_ok[script_top_level_from]
  '''
  pipe_from
    from_clause
      from_item_table
        path_expression
          ident	table1
  
  '''
# ---
